autoemulate/experimental/sensitivity_analysis.py

```diff
@@ -0,0 +1,264 @@
+        if "bounds" not in problem:
+            msg = "problem must contain 'bounds'."
+            raise ValueError(msg)
+
+        if len(problem["names"]) != problem["num_vars"]:
+            msg = "Length of 'names' must match 'num_vars'."
+            raise ValueError(msg)
+        if len(problem["bounds"]) != problem["num_vars"]:
+            msg = "Length of 'bounds' must match 'num_vars'."
+            raise ValueError(msg)
+
+        return problem
+
+    @staticmethod
+    def _generate_problem(X: NumpyLike) -> dict:
+        """
+        Generate a problem definition from a design matrix.
+
+        Parameters
+        ----------
+        X : NumpyLike
+            Simulator input parameter values [n_samples, n_parameters].
+        """
+        if X.ndim == 1:
+            msg = "X must be a 2D array."
+            raise ValueError(msg)
+
+        return {
+            "num_vars": X.shape[1],
+            "names": [f"X{i + 1}" for i in range(X.shape[1])],
+            "bounds": [[X[:, i].min(), X[:, i].max()] for i in range(X.shape[1])],
+        }
+
+    def _sample(self, method: str, N: int) -> NumpyLike:
+        if method == "sobol":
+            # Saltelli sampling
+            return sobol_sample(self.problem, N)
+        if method == "morris":
+            # vanilla Morris (1991) sampling
+            return morris_sample(self.problem, N)
+        msg = f"Unknown method: {method}. Must be 'sobol' or 'morris'."
+        raise ValueError(msg)
+
+    def _predict(self, param_samples: NumpyLike) -> NumpyLike:
+        """
+        Make predictions with emulator for N input samples.
+        """
+
+        param_tensor = self._convert_to_tensors(param_samples)
+        assert isinstance(param_tensor, TensorLike)
+        y_pred = self.emulator.predict(param_tensor)
+
+        # handle types, convert to numpy
+        if isinstance(y_pred, TensorLike):
+            y_pred_np, _ = self._convert_to_numpy(y_pred)
+        elif isinstance(y_pred, DistributionLike):
+            y_pred_np, _ = self._convert_to_numpy(y_pred.mean)
+        else:
+            msg = "Emulator has to return Tensor or Distribution"
+            raise ValueError(msg)
+
+        return y_pred_np
+
+    def _get_output_names(self, num_outputs: int) -> list[str]:
+        """
+        Get the output names from the problem definition or generate default names.
+        """
+        # check if output_names is given
+        if "output_names" not in self.problem:
+            output_names = [f"y{i + 1}" for i in range(num_outputs)]
+        elif isinstance(self.problem["output_names"], list):
+            output_names = self.problem["output_names"]
+        else:
+            msg = "'output_names' must be a list of strings."
+            raise ValueError(msg)
+
+        return output_names
+
+    def run(
+        self,
+        method: str = "sobol",
+        N: int = 1024,
+        conf_level: float = 0.95,
+    ) -> pd.DataFrame | None:
```

Comment:
Related to https://github.com/alan-turing-institute/autoemulate/pull/539/files#r2154672187
```suggestion
    ) -> pd.DataFrame:
```